crates/storage_impl/src/redis/kv_store.rs

```diff
@@ -235,3 +236,61 @@ where
             err
         })
 }
+
+pub enum Op<'a> {
+    Insert,
+    Update(PartitionKey<'a>, &'a str, Option<&'a str>),
+    Find,
+}
+
+pub async fn decide_storage_scheme<'a, T, D>(
+    store: &KVRouterStore<T>,
+    storage_scheme: MerchantStorageScheme,
+    operation: Op<'a>,
+) -> MerchantStorageScheme
+where
+    D: de::DeserializeOwned
+        + serde::Serialize
+        + Debug
+        + KvStorePartition
+        + UniqueConstraints
+        + Sync,
+    T: crate::database::store::DatabaseStore,
+{
+    match operation {
+        Op::Insert => {
+            if store.soft_kill_mode {
+                MerchantStorageScheme::PostgresOnly
+            } else {
+                storage_scheme
+            }
```

Comment:
Let's say we enabled `soft_kill` mode and someone enabled KV in that case data should go to KV. It doesnt happen here

Reply:
if soft kill is enabled , no new data should go to kv, why would we enable kv for a merchant during soft kill

Reply:
Then we should disable that in respective API's. Make sure KV is disabled and dont let anyone change `storage_scheme` for the merchant